pkg/logging/kafka/update.go

```diff
@@ -160,6 +178,50 @@ func (c *UpdateCommand) createInput() (*fastly.UpdateKafkaInput, error) {
 		input.Placement = fastly.String(c.Placement.Value)
 	}
 
+	if c.ParseLogKeyvals.Valid {
+		input.ParseLogKeyvals = fastly.CBool(c.ParseLogKeyvals.Value)
+	}
+
+	if c.RequestMaxBytes.Valid {
+		input.RequestMaxBytes = fastly.Uint(c.RequestMaxBytes.Value)
+	}
+
+	if c.UseSASL.Valid && !c.UseSASL.Value {
+		input.AuthMethod = fastly.String("")
+		input.User = fastly.String("")
+		input.Password = fastly.String("")
+	}
+
+	if c.UseSASL.Valid && c.UseSASL.Value {
+		if c.AuthMethod.Value == "" || c.User.Value == "" || c.Password.Value == "" {
+			return nil, fmt.Errorf("the --auth-method, --username,  and --password flags must be present when using the --use-sasl flag")
+
+		}
+
+		input.AuthMethod = fastly.String(c.AuthMethod.Value)
+
+		if c.User.Valid {
+			input.User = fastly.String(c.User.Value)
+		}
+
+		if c.Password.Valid {
+			input.Password = fastly.String(c.Password.Value)
+		}
+	}
+
+	if !c.UseSASL.Value && (c.AuthMethod.Value != "" || c.User.Value != "" || c.Password.Value != "") {
+		if c.AuthMethod.Value != "" {
+			fmt.Printf("AuthMethod: %s\n", c.AuthMethod.Value)
+		}
+		if c.User.Value != "" {
+			fmt.Printf("AuthMethod: %s\n", c.User.Value)
+		}
+		if c.Password.Value != "" {
+			fmt.Printf("AuthMethod: %s\n", c.Password.Value)
+		}
+		return nil, fmt.Errorf("the --auth-method, --username, and --password options are only valid when the --use-sasl flag is specified")
```

Comment:
Nit: Same comment as in the kafka/create file.  If possible the successful case should fall through the function, so the failures should be at the top of the function.

Reply:
Done